numpy/lib/arraysetops.py

```diff
@@ -307,51 +315,67 @@ def unique(ar, return_index=False, return_inverse=False,
         msg = 'The axis argument to unique is not supported for dtype {dt}'
         raise TypeError(msg.format(dt=ar.dtype)) from e
 
+    # If original array fulfill all requirements:
+    #   - is 2-D
+    #   - has "simple"/unstructred dtype,
+    #     i.e. all elements can be safely interpreted as np.float32 or np.int64
+    #   - original shape is not equal to 0 (in any dimension)
+    # Then this optimization might be used to speed up unique computation along
+    # axis equal to either 0 or 1.
+    if (orig_dim == 2 and len(orig_dtype) < 2 and
+       all(i > 0 for i in orig_shape)):
```

Comment:
I don't understand why orig_ndim matters.  Isn't the array always flattened along all but one dim?